### ISSUE TITLE
[Issue][CSG import]

### ISSUE THREAD

**Issue by Knochi** [FreeCAD/FreeCAD#9268]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Forums discussion

https://forum.freecad.org/viewtopic.php?p=675697&sid=7ed86a4eb1b406f77f166ad2b69288c0#p675697

### Version

0.20 (Release)

### Full version info

```shell
OS: Windows 10 Version 2009
Word size of FreeCAD: 64-bit
Version: 0.20.2.29603 (Git)
Build type: Release
Branch: releases/FreeCAD-0-20
Hash: 930dd9a76203a3260b1e6256c70c1c3cad8c5cb8
Python 3.8.10, Qt 5.15.2, Coin 4.0.1, Vtk 8.2.0, OCC 7.6.3
Locale: German/Germany (de_DE)
Installed mods: 
  * dxf-library
  * kicadStepUpMod 10.15.1
```

### Subproject(s) affected?

None

### Issue description

OpenSCAD code:
```scad
// FreeCad CSG import issue with offset
fudge=0.1;
$fn=50;

body();

module body(D=16,L=30){
  rad=0.5;
  notch=1.0;
  foilThck=0.1;
  foilOpnDia=12.8;
  ringPos=0.15;
  
    rotate_extrude(){     
      difference(){
        offset(rad) 
          difference(){
            translate([0,rad]) square([D/2-rad,L-rad*2]);
            //notch ring
            translate([(D+notch-rad)/2,L*ringPos]) circle(notch+rad);
          }
      //cut away excess from offset
      translate([-(rad+fudge),-fudge]) square([rad+fudge,L+fudge]);
      }
    }
  } 
```

When exporting as CSG and importing to FreeCAD i get the following log messages:

12:45:49 WARNING: Token 'DOT' defined, but not used
13:04:14 WARNING: Token 'DOT' defined, but not used
13:04:14 WARNING: Token 'WORD' defined, but not used
13:04:14 WARNING: There are 2 unused tokens
13:04:14 End processing CSG file
13:05:57 WARNING: Token 'DOT' defined, but not used
13:05:57 WARNING: Token 'WORD' defined, but not used
13:05:57 WARNING: There are 2 unused tokens
13:05:57 Traceback (most recent call last):
File "<string>", line 1, in <module>
File "C:\Users\xxx\AppData\Local\Programs\FreeCAD 0.20\Mod\OpenSCAD\importCSG.py", line 137, in open
processcsg(filename)
File "C:\Users\xxx\AppData\Local\Programs\FreeCAD 0.20\Mod\OpenSCAD\importCSG.py", line 192, in processcsg
result = parser.parse(f.read())
File "C:\Users\xxx\AppData\Local\Programs\FreeCAD 0.20\bin\lib\site-packages\ply\yacc.py", line 333, in parse
return self.parseopt_notrack(input, lexer, debug, tracking, tokenfunc)
File "C:\Users\xxx\AppData\Local\Programs\FreeCAD 0.20\bin\lib\site-packages\ply\yacc.py", line 1120, in parseopt_notrack
p.callable(pslice)
File "C:\Users\xxx\AppData\Local\Programs\FreeCAD 0.20\Mod\OpenSCAD\importCSG.py", line 465, in p_offset_action
if subobj.Shape.Volume == 0 :
<class 'RuntimeError'>: shape is invalid

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by chennes** [comment]
Assuming that this is what the part is supposed to look like:
![Imported part](https://user-images.githubusercontent.com/216920/232176870-4a113fca-89bc-4921-8012-bd877fc1cc49.png)

Then this bug has been resolved in 0.21dev -- if you have the opportunity to do so, please give one of our development builds a shot and see if you can confirm. I will close this for now, but please feel free to re-open if the dev version does not fix it on your machine.

**Comment by Knochi** [comment]
Can confirm this works with 0.21.0
